theano/gpuarray/multinomial.py

```diff
@@ -227,23 +227,27 @@ def c_code_cache_version(self):
         return (1,)
 
 
-@register_opt()
+@register_opt('fast_compile')
 @op_lifter([theano.sandbox.multinomial.MultinomialFromUniform])
-def local_gpua_multinomial(node, context_name):
+@register_opt2([theano.sandbox.multinomial.MultinomialFromUniform], 'fast_compile')
+def local_gpua_multinomial(op, context_name, inputs, outputs):
     # TODO : need description for function
 
-    if len(node.inputs) == 2:
-        p, u = node.inputs
+    if len(inputs) == 2:
+        p, u = inputs
         n_samples = 1
     else:
-        p, u, n_samples = node.inputs
+        p, u, n_samples = inputs
     try:
         if get_scalar_constant_value(n_samples) != 1:
             return None
     except NotScalarConstantError:
         return None
-    m, = node.outputs
+
+    node = op.make_node(*inputs)
+    outputs = node.outputs
+    m, = outputs
```

Comment:
Yes! We initially didn't have outputs passed as parameter. The time when i added that, i didn't notice this. Will make the changes